edgedb/con_utils.py

```diff
@@ -326,6 +369,76 @@ def _validate_user(user):
     return user
 
 
+def _parse_iso_duration(string: str):
+    match = ISO_DURATION_RE.match(string)
+    if match is None:
+        raise ValueError(f"invalid duration {string!r}")
+
+    return (
+        3600 * float(match.group('hours') or '0') +
+        60 * float(match.group('minutes') or '0') +
+        float(match.group('seconds') or '0')
+    )
+
+
+def _parse_human_duration_unit(re, string):
+    number = None
+    match = re.search(string)
+    if match:
+        number = match.group('number')
+        tail = match.group('tail')
+        string = re.sub(tail, string, count=1)
+
+    return number, string
+
+
+def _parse_human_duration(string: str):
+    hour, string = _parse_human_duration_unit(HUMAN_DURATION_HOUR_RE, string)
+    minute, string = _parse_human_duration_unit(
+        HUMAN_DURATION_MINTUE_RE, string)
+    second, string = _parse_human_duration_unit(
+        HUMAN_DURATION_SECOND_RE, string)
+    ms, string = _parse_human_duration_unit(HUMAN_DURATION_MS_RE, string)
+    us, string = _parse_human_duration_unit(HUMAN_DURATION_US_RE, string)
+
+    if string.strip() != '':
+        raise ValueError(f'invalid duration {string!r}')
+
+    no_value = (
```

Comment:
This condition seems too forgiving to me.  It should be an error if there is _any_ unrecognized text in the passed value to avoid confusion and bugs.

Reply:
Nevermind, I see the extra check.